src/wp-includes/option.php

```diff
@@ -853,7 +862,7 @@ function update_option( $option, $value, $autoload = null ) {
 			} else {
 				wp_cache_set( $option, $serialized_value, 'options' );
 			}
-		} elseif ( 'yes' === $update_args['autoload'] ) {
+		} elseif ( 'on' === $update_args['autoload'] || 'auto-on' === $update_args['autoload'] || 'auto' === $update_args['autoload'] || 'yes' === $update_args['autoload'] ) {
```

Comment:
We can use the new function here too.
```suggestion
		} elseif ( in_array( $update_args['autoload'], wp_autoload_values_to_autoload(), true ) ) {
```